### PR TITLE
ENH: Import/Export Plot Config

### PR DESCRIPTION
**Pull request by aksharsarvesh**
Opening the Archive Viewer, changing plot config, then saving it to a file requires that the plot config can be saved in a dictionary style data structure. This to_dict() function does that for us.

### CODE REVIEW

File: pydm/widgets/baseplot.py [hunk 1]
```diff
@@ -739,6 +739,21 @@ def __init__(
         if utilities.is_qt_designer():
             self.installEventFilter(self)
 
+    def to_dict(self) -> OrderedDict:
```
Comment (YektaY) [inline pydm/widgets/baseplot.py:742]: can you add docstring to this method?

File: pydm/widgets/timeplot.py [hunk 2]
```diff
@@ -492,6 +492,11 @@ def __init__(
         self.auto_scroll_timer = QTimer()
         self.auto_scroll_timer.timeout.connect(self.auto_scroll)
 
+    def to_dict(self) -> OrderedDict:
```
Comment (YektaY) [inline pydm/widgets/timeplot.py:495]: can you add a docstring to this method as well?

**Review by YektaY — COMMENTED** [review]
I added a couple notes on adding doctrings

**Review by YektaY — APPROVED** [review]

**Comment by jbellister-slac** [comment]
This one I think I need to be pointed to the corresponding Archive Viewer code to understand. The majority of these options are already stored in the underlying `.ui` file of base plots. Is the plot for the Archive Viewer using custom save/load functionality rather than a`.ui' file?

**Comment by aksharsarvesh** [comment]
@zdomke may be able to answer this better than me, but yes - the goal of this is to be able to import and export whole plots with all configurations and settings. Analogous to how ArchivePlotCurveItem (and now FormulaCurveItem) has a to_dict function to store things like which color the curve is set to, its axis, and other settings, we also want to save things like plot title, whether Xgrid/third is being shown, etc. rather than storing this in a .ui file (if that’s what you’re suggesting) Zach and I opted to have an import/export file that stores a barebones dictionary of these settings for easy storage and we have functionality to read in such a dictionary and recreate the graph from the ground up.

**Comment by zdomke** [comment]
@jbellister-slac This information is stored in the `.ui` file but, at least in our project, the user has the option to change all of these properties. Once they've been changed they'll no longer align with the `.ui` file. The Archive Viewer is using a custom save/load function, and we'd like to get the plot's state on request. We figured that a `to_dict` function similar to those of the plot curve items would be a good way of doing that.

For our needs we could move this to the Archive Viewer code, but we thought that it was generalized enough that other pydm users might get use of it too. Please let us know if you'd like us to move this to our side of things.

**Comment by jbellister-slac** [comment]
Ok, thank you both for the context! I understand better now

**Review by jbellister-slac — APPROVED** [review]

**Pull request merged**